src/main/java/ch/njol/skript/conditions/CondEntityIsIn.java

```diff
@@ -0,0 +1,92 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.conditions;
+
+import ch.njol.skript.conditions.base.PropertyCondition;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Condition;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import org.bukkit.entity.Entity;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+
+@Name("Is in Rain/Lava/Water/BubbleColumn")
+@Description("Checks whether an entity is in Rain/Lava/Water/BubbleColumn")
+@Examples({"player is in rain",
+		"player is in water",
+		"player is in lava",
+		"player is in bubble column",
+		"player is in water or rain",
+		"player is in water or bubble column",
+		"player is in water or rain or bubble column"})
+@Since("INSERT VERSION")
+public class CondEntityIsIn extends Condition {
```

Comment:
i don't really know if it's good to register this all in a single condition either

Reply:
> i don't really know if it's good to register this all in a single condition either


I didn't want to add many new classes for each one, it doesn't make sense to have all that extra size

Reply:
Also, these all are related `water, rain, lava and bubble column` however, I was about to make another PR for multiple conditions but they are not related to each other so I seperated them.

Reply:
I think it would be okay as the conditions are related - it seems a little much to me to have conditions like
CondIsInWater, CondIsInRain, CondIsInLava, etc.

Reply:
I dunno, I don't really agree because the only thing sticking them all into a single class is doing is adding a bunch of complexity such as the string builder in init, all the parse marks, all the static final fields, and the `getCondName` method. In current Skript, similar conditions are still split up. Take `CondIsOnGround` and `CondIsFlying` for example.

Reply:
Take a look at how simple [CondIsFlying](https://github.com/SkriptLang/Skript/blob/master/src/main/java/ch/njol/skript/conditions/CondIsFlying.java) is for comparison

Reply:
Gotcha, with my respect I don't mind splitting them up but it is just not necessary because they're all related as said above and we always try to improve over time.

The examples you gave were added long time ago and they are very simple indeed same for the current feature if splitted up they will be simpler but just not needed IMO only because they're related. However, if more devs agree to split them up I got no problem doing that since this is just an open discussion.

Reply:
> 
> 
> I dunno, I don't really agree because the only thing sticking them all into a single class is doing is adding a bunch of complexity such as the string builder in init, all the parse marks, all the static final fields, and the `getCondName` method. In current Skript, similar conditions are still split up. Take `CondIsOnGround` and `CondIsFlying` for example.

I definitely see what you are saying - I guess to me it most makes sense since there are the methods/syntax like "in water or rain"

The method checks seem a little excessive to me though - were these methods not all added in the same version?

Reply:
They where added at the same time in 1.16 in paper (lava, rain, bubble column) and spigot 1.16 (water)

I don't like the `in water or rain` either because it may confuse the users due to the `or` which I mentioned in the PR description but even tho if I split them I will keep the syntaxes for `in water or rain` and others unless you guys don't want them.

Reply:
maybe we could just make use of better patterns for these, like:
```
%entities% (is|are) in (water|lava)
%entities% (is|are) soaked [by %-soaktype%]
```
where soaktype can be rain, water or bubble column, or all of them if not specified. Maybe a bit too complex for a check this simple though, I'd like to see input on this

Reply:
> maybe we could just make use of better patterns for these, like:
> 
> ```
> %entities% (is|are) in %soaktype%
> %entities% (is|are) soaked [by %-soaktype%]
> %entities% (is|are) wet # would be the isInWaterOrRainOrBubbleColumn method, such a bad name for a method
> ```
> 
> where soaktype can be rain, water, lava or bubble column, or all of them if not specified. Maybe a bit too complex for a check this simple though, I'd like to see input on this

I think `%entities% (is|are) wet` for `#isInWaterOrRainOrBubbleColumn #isInWaterOrRain #isInWaterOrBubbleColumn` could be useful since they all means wet however, for the `is soaked by %soaktype%` could be used as well but it wouldn't replace the current methods IMO only the multiple checks (weird, long method names) because `is in water/lava/rain/bubble column` is easier to come up with specailly for basic english users